src/client/App.tsx

```diff
@@ -1,13 +1,60 @@
-import "./App.css";
-
-import { useEffect, useState } from "react";
+import { ChangeEvent, useState } from "react";
+import { generate } from "./api";
 
 function App() {
+  const [startDate, setStartDate] = useState(new Date(2020, 7, 20, 0, 0, 0));
+  const [endDate, setEndDate] = useState(new Date(2020, 7, 21, 0, 0, 0));
+  const [commits, setCommits] = useState(10);
+  const [frequency, setFrequency] = useState(50);
+
+  const onChangeStartDate = (e: ChangeEvent<HTMLInputElement>) => {
+    const values = e.target.value.split("-");
+    setStartDate(new Date(Number(values[0]), Number(values[1]) - 1, Number(values[2]), 0, 0, 0));
+  }
+  const onChangeEndDate = (e: ChangeEvent<HTMLInputElement>) => {
+    const values = e.target.value.split("-");
+    setEndDate(new Date(Number(values[0]), Number(values[1]) - 1, Number(values[2]), 0, 0, 0));
+  }
+  const onChangeCommits = (e: ChangeEvent<HTMLInputElement>) => {
+    setCommits(Number(e.target.value));
+  }
+  const onChangeFrequency = (e: ChangeEvent<HTMLInputElement>) => {
+    setFrequency(Number(e.target.value));
+  }
+  const onClickGenerate = async () => {
+    if (startDate > endDate || commits < 0 || frequency < 0) {
+      alert("Invalid data");
+    }
```

Comment:
Also endDate shouldn't be after today for validation.
And commit, frequency need to have maximum value limit for validation